api.ts

```diff
@@ -1372,6 +1376,24 @@ export interface DynamicSyncTableOptions<
    * this sync table (including autocomplete formulas).
    */
   connectionRequirement?: ConnectionRequirement;
+  /**
+   * If true, when subsequent syncs discover new schema fields, these fields will not automatically be
+   * added as new columns on the table. The user can still manually add columns for these new fields.
+   * This only applies to tables that use dynamic schemas.
+   *
+   * When tables with dynamic schemas are synced, the {@link getSchema} formula is run each time,
+   * which may return a schema that is different than that from the last sync. The default behavior
+   * is that any schema fields that are new in this sync are automatically added as new columns,
+   * so they are apparent to the user. However, in rare cases when schemas change frequently,
+   * this can cause the number of columns to grow quickly and become overwhelming. Setting this
+   * value to true leaves the columns unchanged and puts the choice of what columns to display
+   * into the hands of the user.
+   */
+  doNotAddNewSyncColumnsByDefault?: boolean;
```

Comment:
This field name is a bit verbose, and the `doNot` prefix is uncommon. What about something like:

```ts
disableNewColumnsOnSync: true
```